### PR TITLE
CI: dynamic testing

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Addresses https://github.com/PyTorchLightning/metrics/issues/678#issuecomment-1013232707

## Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: .github/workflows/ci_test-full.yml [hunk 1]
```diff
@@ -13,10 +13,33 @@ concurrency:
   cancel-in-progress: ${{ ! (github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/heads/release/')) }}
 
 jobs:
-  pytest:
+  # ToDo: consider unify in single workflow and distribute to all others
```
Comment (Borda) [inline .github/workflows/ci_test-full.yml:16]:
```suggestion
  # ToDo: consider unifying in a single workflow and distributing outputs to all others depending ones
```

File: .github/workflows/ci_test-conda.yml [hunk 2]
```diff
@@ -12,16 +12,40 @@ concurrency:
   cancel-in-progress: ${{ ! (github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/heads/release/')) }}
 
 jobs:
+  # ToDo: consider unify in single workflow and distribute to all others
```
Comment (Borda) [inline .github/workflows/ci_test-conda.yml:15]:
```suggestion
  # ToDo: consider unifying in a single workflow and distributing outputs to all others depending ones
```

**Comment by SkafteNicki** [comment]
Should we have some tests that the assistant is doing what it is supposed to do?

**Comment by Borda** [comment]
> Should we have some tests that the assistant is doing what it is supposed to do?

I had checked it on other pr #764 and if we merge it we can also test it on #773

**Review by SkafteNicki — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Comment by Borda** [comment]
seems to be working as expected... run just text-domain for this related deprecation pruning
image
https://github.com/PyTorchLightning/metrics/pull/773/checks

**Comment by Borda** [comment]
or the similar just fro improving one classification metric:
https://github.com/PyTorchLightning/metrics/pull/769/checks

**Comment by Borda** [comment]
image

**Pull request merged**